modules/http/http-worker.c

```diff
@@ -520,8 +522,23 @@ _curl_perform_request(HTTPDestinationWorker *self, HTTPLoadBalancerTarget *targe
             evt_tag_str("url", target->url));
 
   curl_easy_setopt(self->curl, CURLOPT_URL, target->url);
+  if (owner->message_compression != CURL_COMPRESSION_UNCOMPRESSED)
+    {
+      if(compressor_compress(self->compressor, self->request_body_compressed, self->request_body))
```

Comment:
[optional]

I think we should add a `g_string_truncate(self->request_body_compressed, 0)` to `_reinit_request_body()`.
The `request_body` is also truncated there, and these 2 variables should change together.

Reply:
Thanks!